### ISSUE TITLE
Unable to invalidate conj_key generated by read against UUID field

### ISSUE THREAD

**Issue by andytzeng** [Suor/django-cacheops#312]
To Team,

Here is my example model
```
class Device(models.Model):
    uid = models.UUIDField()
    brand = models.CharField(max_length=64, null=True)
    model = models.CharField(max_length=64, null=True)
    created = models.DateTimeField(auto_now_add=True)
```

The issue happen that if query item by UUID field as follow:
```
Device.objects.get(uid='584d13481e6b4049b5b7b6c1075847c8')
```

When I observed generated keys in Redis, it generated the following conj key
```
cacheops_conj:user_device:uid=584d13481e6b4049b5b7b6c1075847c8
```

I have tried to purge Redis and query by dash format or UUID object
```
Device.objects.get(uid='584d1348-1e6b-4049-b5b7-b6c1075847c8')
# or
# Device.objects.get(uid=uuid.UUID('584d13481e6b4049b5b7b6c1075847c8'))
# Device.objects.get(uid=uuid.UUID('584d1348-1e6b-4049-b5b7-b6c1075847c8'))
```
The conj key generated will have dash inside as below:
```
cacheops_conj:user_device:uid=584d1348-1e6b-4049-b5b7-b6c1075847c8
```
When I used invalidate command or API call to purge the cache, I found case 1 (no dash) cannot be purged and case 2 works.
Similarly, when save() is invoked and invalidate the cache, only dash-format will works.

After diging the code, I found ```invalidate_dict``` generated keys by invoking **str** on UUID field, the format always contains dash.

Please consider normalizing the two generating keys.
I expected that no matter how I query the object (32-byte characters or UUID format with dashes), the cached object should be able to be invalidated.

I found the issue similar to #15 , but different data type.

**Comment by Suor** [comment]
Can you write a test for it?

**Comment by andytzeng** [comment]
> Can you write a test for it?

would you like PR a test against the scenario?

**Comment by Suor** [comment]
Yes

>

**Comment by andytzeng** [comment]
Sure, I can do that. :)

**Comment by Suor** [comment]
Cool. The process is described at the bottom of the README.

**Comment by andytzeng** [comment]
PR has submitted for test case!. Thanks.

**Comment by Suor** [comment]
Good, I'll look into it.

пт, 14 дек. 2018 г., 16:07 Andy Tzeng notifications@github.com:

**Comment by Suor** [comment]
This is fixed in Django 3.0.
